### PR TITLE
Updated NV events.py

### PR DESCRIPTION
**Pull request by a-cat-named-snowball**
Switched over to scraping this page: https://www.leg.state.nv.us/App/NELIS/REL/82nd2023/Meetings

### CODE REVIEW

File: scrapers/nv/events.py [hunk 1]
```diff
@@ -1,82 +1,167 @@
...
+            date = datetime.datetime.strptime(date, "%B %d, %Y %I:%M %p")
+
+            # Third element contains: a list of locations
+            locations = XPath(".//li/text()").match(i[2])
+
+            # Build event, scrape bill ids from agenda, and yield it
+            event = Event(
+                start_date=self._TZ.localize(date),
+                name=title,
+                location_name=locations[0],
```
Comment (chrisyamas) [inline scrapers/nv/events.py:150]: Can you add a replace statement to remove the no-break-space that is currently showing up in the location name between "Carson" and "City" on any events? `"\u00a0"` needs to be replaced with single space `" "`.

**Review by chrisyamas — CHANGES_REQUESTED** [review]
Just the one change needed on no-break-space. Otherwise, looks great! Nice job, especially considering the amount of modification that was needed here.

File: scrapers/nv/events.py [hunk 2]
```diff
@@ -1,82 +1,167 @@
...
+
+class AgendaStartTime(PdfPage):
+    def process_page(self):
+        # Simplify text to make regex simpler
+        self.text = self.text.replace(". ", "").replace(".", "")
+
+        start = start_time_re.search(self.text)
+        if not start:
+            # Default to 12am if no start time is found
+            yield "12:00 am"
```
Comment (johnseekins) [inline scrapers/nv/events.py:27]: A single item generator is no different from a regular `return`. Probably just return values here.

File: scrapers/nv/events.py [hunk 3]
```diff
@@ -1,82 +1,167 @@
...
+        # Simplify text to make regex simpler
+        self.text = self.text.replace(". ", "").replace(".", "")
+
+        # Find all bill ids
+        bills = bills_re.findall(self.text)
+
+        # Format bills before yielding
+        formatted_bills = set()
+        for alpha, num in bills:
+            formatted_bills.add(f"{alpha.upper()} {num}")
```
Comment (johnseekins) [inline scrapers/nv/events.py:44]: Clever way to dodge duplicates or just a more complicated way to say:

```
        for alpha, num in bills:
            yield f"{alpha.upper()} {num}")
```
?

File: scrapers/nv/events.py [hunk 4]
```diff
@@ -1,82 +1,167 @@
...
+        # divs and places them into the meetings variable
+        for i in self.root.getchildren():
+            if i.tag == "div":
+                cur_row.append(i)
+            elif i.tag == "hr":
+                # Only add if all three elements are present - missing elements
+                # indicate that this section is cancelled or floor meeting
+                if len(cur_row) == 3:
+                    meetings.append(cur_row)
+                cur_row = []
```
Comment (johnseekins) [inline scrapers/nv/events.py:97]: Are you _sure_ you want to empty this higher scoped variable here?

File: scrapers/nv/events.py (hunk 2 above)
Comment (johnseekins) [reply scrapers/nv/events.py:27]: Oh. Except this needs `yield` to fulfill the object's contract, right? Never mind.

File: scrapers/nv/events.py (hunk 3 above)
Comment (a-cat-named-snowball) [reply scrapers/nv/events.py:44]: My intent was to prevent duplicates

Comment (johnseekins) [reply scrapers/nv/events.py:44]: Cool. Just making sure. Thanks!

File: scrapers/nv/events.py (hunk 4 above)
Comment (a-cat-named-snowball) [reply scrapers/nv/events.py:97]: Yes - If there aren't exactly 3 elements in cur_row when a <hr> is found, then cur_row should be emptied without adding it to the meetings list.

Comment (johnseekins) [reply scrapers/nv/events.py:97]: Seems reasonable. Can you add a quick note explaining why you're emptying the variable? I could see someone coming back to this later and being confused.

File: scrapers/nv/events.py (hunk 2 above)
Comment (a-cat-named-snowball) [reply scrapers/nv/events.py]: Thanks, fixed with https://github.com/openstates/openstates-scrapers/pull/4540/commits/3fd4f2a413896dffb5f64f83bc51d257ceaac8f0

**Review by chrisyamas — APPROVED** [review]

**Pull request merged**